diff --git a/openssl-compat.h b/openssl-compat.h
index 69afc716..63f24493 100644
--- a/openssl-compat.h
+++ b/openssl-compat.h
@@ -1,7 +1,7 @@
 #ifndef OPENSSL_COMPAT_H
 #define OPENSSL_COMPAT_H
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined LIBRESSL_VERSION_NUMBER
 
 static inline BIO_METHOD *BIO_meth_new(int type, const char *name)
 {
@@ -30,6 +30,8 @@ static inline BIO_METHOD *BIO_meth_new(int type, const char *name)
 
 #define TLS_method SSLv23_method
 
+#define ASN1_STRING_get0_data(x) ASN1_STRING_data(x)
+
 #endif /* OPENSSL_VERSION_NUMBER < 0x10100000L */
 
 #endif /* OPENSSL_COMPAT_H */
diff --git a/sample/openssl_hostname_validation.c b/sample/openssl_hostname_validation.c
index 40312f2e..5b1bef45 100644
--- a/sample/openssl_hostname_validation.c
+++ b/sample/openssl_hostname_validation.c
@@ -41,6 +41,7 @@ SOFTWARE.
 
 #include <openssl/x509v3.h>
 #include <openssl/ssl.h>
+#include "openssl-compat.h"
 #include <string.h>
 
 #include "openssl_hostname_validation.h"
