From f99fc63082147d41540aefc4268aa0c9df37900f Mon Sep 17 00:00:00 2001
From: Alex Catarineu <a.catarineu@gmail.com>
Date: Sun, 17 Sep 2017 18:46:08 +0200
Subject: [PATCH 2/3] Remove asm in __warn_references

---
 include/compat/sys/types.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/compat/sys/types.h b/include/compat/sys/types.h
index 701b77a..b34cd04 100644
--- a/include/compat/sys/types.h
+++ b/include/compat/sys/types.h
@@ -56,7 +56,7 @@ typedef SSIZE_T ssize_t;
 #define __STRING(x) #x
 #endif
 
-#if defined(__GNUC__)  && defined (HAS_GNU_WARNING_LONG)
+#if defined(__GNUC__)  && defined (HAS_GNU_WARNING_LONG) && !defined(OPENSSL_NO_ASM)
 #define __warn_references(sym,msg)          \
   __asm__(".section .gnu.warning." __STRING(sym)  \
          " ; .ascii \"" msg "\" ; .text");
-- 
2.11.0

